acvm-repo/acvm/tests/solver.rs

```diff
@@ -722,3 +724,531 @@ fn memory_operations() {
+        let expected_results = drop_use_constant(&xs);
+        let add_results = bigint_solve_binary_op(bigint_add_op(), modulus.clone(), xs, ys.clone());
+        let add_bigint: Vec<_> = add_results.into_iter().map(|x| (x, false)).collect();
+        let results = bigint_solve_binary_op(bigint_sub_op(), modulus, add_bigint, ys);
+
+        prop_assert_eq!(results, expected_results)
+    }
+
+    // TODO(https://github.com/noir-lang/noir/issues/5579)
+    #[test]
+    #[should_panic(expected = "Test failed: assertion failed: `(left == right)")]
+    fn bigint_sub_add((xs, ys, modulus) in bigint_pair_with_modulus()) {
+        let expected_results = drop_use_constant(&xs);
+        let sub_results = bigint_solve_binary_op(bigint_sub_op(), modulus.clone(), xs, ys.clone());
+        let add_bigint: Vec<_> = sub_results.into_iter().map(|x| (x, false)).collect();
+        let results = bigint_solve_binary_op(bigint_add_op(), modulus, add_bigint, ys);
+
+        prop_assert_eq!(results, expected_results)
+    }
+
+    // TODO(https://github.com/noir-lang/noir/issues/5579): Fails on 49, see bigint_add_zero_l_single_case_49
+    #[test]
+    #[should_panic(expected = "Test failed: assertion failed: `(left == right)`")]
+    fn bigint_div_mul((xs, ys, modulus) in bigint_pair_with_modulus()) {
+        let expected_results = drop_use_constant(&xs);
+        let div_results = bigint_solve_binary_op(bigint_div_op(), modulus.clone(), xs, ys.clone());
+        let div_bigint: Vec<_> = div_results.into_iter().map(|x| (x, false)).collect();
+        let results = bigint_solve_binary_op(bigint_mul_op(), modulus, div_bigint, ys);
+
+        prop_assert_eq!(results, expected_results)
+    }
+
+    // TODO(https://github.com/noir-lang/noir/issues/5579): Fails on 49, see bigint_add_zero_l_single_case_49
+    #[test]
+    #[should_panic(expected = "Test failed: assertion failed: `(left == right)`")]
+    fn bigint_mul_div((xs, ys, modulus) in bigint_pair_with_modulus()) {
+        let expected_results = drop_use_constant(&xs);
+        let mul_results = bigint_solve_binary_op(bigint_mul_op(), modulus.clone(), xs, ys.clone());
+        let mul_bigint: Vec<_> = mul_results.into_iter().map(|x| (x, false)).collect();
+        let results = bigint_solve_binary_op(bigint_div_op(), modulus, mul_bigint, ys);
+
+        prop_assert_eq!(results, expected_results)
+    }
+}
+
+// TODO(https://github.com/noir-lang/noir/issues/5579):
+// this test is redundant with bigint_add_zero_l, but may be useful for
+// debugging. It can be removed once bigint_add_zero_l is
+// passing because proptest automatically retries previous failures first.
+#[test]
+#[should_panic(expected = "assertion `left == right` failed")]
+fn bigint_add_zero_l_single_case_49() {
+    let modulus = &allowed_bigint_moduli()[0];
+    let mut xs: Vec<_> = modulus.iter().map(|_| (FieldElement::zero(), false)).collect();
+    xs[modulus.len() - 1] = (FieldElement::from(49u128), false);
```

Comment:
how do you know whether this overflow the modulus or not?

Reply:
I'm not sure: the last byte in `allowed_bigint_moduli` is `0x30` so perhaps it does overflow?

Reply:
then yes it overflows, you should ensure that the inputs do not overflow

Reply:
@guipublic I added a simple byte-by-byte clamp to ensure the maximum is the modulus and most of the tests are now passing.